File: Serene/Serene.Script/Imports/FormContexts/Membership.ChangePasswordForm.cs

namespace Serene.Membership
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class ChangePasswordForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Membership.ChangePassword";
    
        public ChangePasswordForm(string idPrefix) : base(idPrefix) {}
    
        public PasswordEditor OldPassword { [InlineCode("{this}.w('OldPassword', Serenity.PasswordEditor)")] get; private set; }
        public PasswordEditor NewPassword { [InlineCode("{this}.w('NewPassword', Serenity.PasswordEditor)")] get; private set; }
        public PasswordEditor ConfirmPassword { [InlineCode("{this}.w('ConfirmPassword', Serenity.PasswordEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Imports/ServerTypings/Membership.ChangePasswordForm.ts
namespace Serene.Membership {
    export interface ChangePasswordForm {
        OldPassword: Serenity.PasswordEditor;
        NewPassword: Serenity.PasswordEditor;
        ConfirmPassword: Serenity.PasswordEditor;
    }

    export class ChangePasswordForm extends Serenity.PrefixedContext {
        static formKey = 'Membership.ChangePassword';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!ChangePasswordForm.init)  {
                ChangePasswordForm.init = true;

                var s = Serenity;
                var w0 = s.PasswordEditor;

                Q.initFormType(ChangePasswordForm, [
                    'OldPassword', w0,
                    'NewPassword', w0,
                    'ConfirmPassword', w0
                ]);
            }
        }
    }
}

